core/templates/pages/landing-pages/topic-landing-page/topic-landing-page.controller.ts

```diff
@@ -51,14 +51,25 @@ angular.module('oppia').directive('topicLandingPage', [
           var getImageData = function(index) {
             var imageKey = 'image_' + index;
             if (landingPageData[imageKey]) {
-              var imagePath = UrlInterpolationService.interpolateUrl(
+              var imagePathPng = UrlInterpolationService.interpolateUrl(
                 angular.copy(assetsPathFormat), {
                   subject: ctrl.subject,
                   topic: topic,
-                  file_name: landingPageData[imageKey].file_name
+                  file_name: landingPageData[imageKey].file_name_png
+                });
+              var imagePathWebp = UrlInterpolationService.interpolateUrl(
+                angular.copy(assetsPathFormat), {
+                  subject: ctrl.subject,
+                  topic: topic,
+                  file_name: landingPageData[imageKey].file_name_webp
                 });
               return {
-                src: UrlInterpolationService.getStaticImageUrl(imagePath),
+                src_png: UrlInterpolationService.getStaticImageUrl(
+                  imagePathPng
+                ),
+                src_webp: UrlInterpolationService.getStaticImageUrl(
+                  imagePathWebp
+                ),
```

Comment:
Wouldn't it be easier to have one field in the dict for the filename and then append appropriate extension?

Reply:
I was not sure as to how to append the extension in the scope variable in the html file..so I did this.

Reply:
You don't need to do that in HTML, just do something like
```javascript
src_png: UrlInterpolationService.getStaticImageUrl(
    imagePath + '.png'
),
src_webp: UrlInterpolationService.getStaticImageUrl(
    imagePath + '.webp'
),
```

Reply:
Okay! so i made the changes in the controller, but we still need to have two variables for png and webp.